python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -62,19 +66,52 @@ _DECIMAL_AGGS = {"COUNT", "SUM", "ARGMIN", "ARGMAX", "MIN", "MAX", "NUNIQUE",
 # workaround for https://github.com/cython/cython/issues/3885
 ctypedef const scalar constscalar
 
+
+cdef _agg_result_from_columns(
+    vector[libcudf_groupby.aggregation_result]& c_result_columns,
+    column_orders,
+    n_input_columns
+):
+    """Construct the list of result columns from libcudf result. The result
+    contains the same number of lists as the number of input columns. Result
+    for an input column that has no applicable aggregations is an empty list.
+    """
+    n_res_cols = c_result_columns.size()
+    # Note we are constructing a list of python columns in the reverse order
+    # to the result in libcudf. This is because the line below requires popping
+    # from this list in FIFO order. In python, popping from the end of the list
+    # is O(1) but popping from head is O(N).
+    result_columns = [
+        [
+            Column.from_unique_ptr(
+                move(c_result_columns[n_res_cols - i - 1].results[j])
+            ) for j in range(
+                c_result_columns[n_res_cols - i - 1].results.size()
+            )
+        ] for i in range(n_res_cols)
+    ]
```

Comment:
Sure. Your suggestion singles out the reversed access characteristic of the algorithm. But we might need to declare `c_result_i` as a reference to avoid copy.